analyzer_plugin/lib/src/view_extraction.dart

```
@@ -185,6 +188,50 @@ class ViewExtractor extends AnnotationProcessorMixin {
       }
     }
   }
+
+  void findExports(
+      ast.Annotation annotation, List<ExportedIdentifier> exports) {
+    // ignore: omit_local_variable_types
+    final ast.Expression listExpression =
+        getNamedArgument(annotation, 'exports');
+    final alreadyExported = new HashSet<String>();
+    if (listExpression is ast.ListLiteral) {
+      // ignore: omit_local_variable_types
+      for (final ast.Expression item in listExpression.elements) {
+        if (item is ast.Identifier) {
+          // Be wary: item.name includes prefix for PrefixedIdentifier.
+          var name = item.name;
+
+          // Check dupes BEFORE separating the identifier & prefix
+          if (alreadyExported.contains(name)) {
+            errorReporter.reportErrorForNode(
+                AngularWarningCode.DUPLICATE_EXPORT, item, [name]);
+          }
+          alreadyExported.add(name);
+
+          // Separate the identifier & prefix
+          var prefix = '';
+          if (item is ast.PrefixedIdentifier) {
```

not all prefixed identifiers are allowed. you should report an error if the Prefix is not a library prefix (ie, `MyClass.staticField` is not allowed, you must export `MyClass` instead). the angular compiler will report an error in this case